### ISSUE TITLE
Abi error when type name is same as action_name or table_name

### ISSUE THREAD

**Issue by clayop** [EOSIO/eos#722]
This is an abi created by `eoscpp -n hello`
~~~
{
  "types": [{
      "new_type_name": "account_name",
      "type": "name"
    }
  ],
  "structs": [{
      "name": "transfer",
      "base": "",
      "fields": {
        "from": "account_name",
        "to": "account_name",
        "amount": "uint64"
      }
    },{
      "name": "account",
      "base": "",
      "fields": {
        "account": "name",
        "balance": "uint64"
      }
    }
  ],
  "actions": [{
      "action": "transfer",
      "type": "transfer"
    }
  ],
  "tables": [{
      "table": "account",
      "type": "account",
      "index_type": "i64",
      "key_names" : ["account"],
      "key_types" : ["name"]
    }
  ]
}
~~~

After publishing the contract bt `eosc set contract tester hello.wast hello.abi`, I obtained the abi by `eosc get code -a hello.abi tester`. And the abi file has changed (note the `"action_name"` and `"table_name"`)

~~~
{
  "types": [{
      "new_type_name": "account_name",
      "type": "name"
    }
  ],
  "structs": [{
      "name": "transfer",
      "base": "",
      "fields": {
        "from": "account_name",
        "to": "account_name",
        "amount": "uint64"
      }
    },{
      "name": "account",
      "base": "",
      "fields": {
        "account": "name",
        "balance": "uint64"
      }
    }
  ],
  "actions": [{
      "action_name": "",
      "type": "transfer"
    }
  ],
  "tables": [{
      "table_name": "",
      "index_type": "i64",
      "key_names": [
        "account"
      ],
      "key_types": [
        "name"
      ],
      "type": "account"
    }
  ]
}
~~~

If I modify the `action_name` to `"trnsfer" which is not same as ["actions"][0]["type"] and `table_name` to `"tabl"`, this bug does not happen.

**Comment by elmato** [comment]
@heifner This has been fixed by #872?

**Comment by heifner** [comment]
Possibly, do you have time to verify?

**Comment by elmato** [comment]
Fixed by #872
